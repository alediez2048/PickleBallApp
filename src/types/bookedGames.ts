// BookedGame type for booked games feature
// All fields must match the structure used in Supabase and the app

export interface BookedGame {
  id: string;
  game_id: string;
  date: string;
  time: string;
  court_name: string;
  location_id: string | null;
  skill_rating: number;
  price: number;
  user_id: string;
  user_info:{} | any
  status: "upcoming" | "completed" | "cancelled";
}
