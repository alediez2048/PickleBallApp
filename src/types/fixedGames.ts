import type { SkillLevel } from "@/types/skillLevel";

// Types for Fixed Games based on the SQL schema
export type DayOfWeek =
  | 'Sunday'
  | 'Monday'
  | 'Tuesday'
  | 'Wednesday'
  | 'Thursday'
  | 'Friday'
  | 'Saturday';

export type FixedGameStatus = 'active' | 'inactive';

export interface FixedGame {
  id: string;
  title: string;
  description?: string;
  day_of_week: DayOfWeek;
  start_time: string; // 'HH:MM:SS' format
  duration_minutes: number;
  location_id: string;
  location?: any; // Add location for joined data
  host: any; // JSON object, can be typed further if needed
  max_players: number;
  skill_level: SkillLevel;
  price: number;
  image_url?: string;
  status: FixedGameStatus;
  created_at: string;
  updated_at: string;
}

export type FixedGameInput = Omit<FixedGame, 'id' | 'created_at' | 'updated_at'>;
