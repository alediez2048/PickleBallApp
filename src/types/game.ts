export const SkillLevel = {
  Beginner: 'Beginner',
  Intermediate: 'Intermediate',
  Advanced: 'Advanced',
  Open: 'Open',
} as const;

export type SkillLevel = typeof SkillLevel[keyof typeof SkillLevel];

export const GameStatus = {
  Upcoming: 'Upcoming',
  InProgress: 'InProgress',
  Completed: 'Completed',
  Cancelled: 'Cancelled',
} as const;

export type GameStatus = typeof GameStatus[keyof typeof GameStatus];

export interface Location {
  id: string;
  name: string;
  address: string;
  city: string;
  state: string;
  zipCode: string;
  coordinates: {
    latitude: number;
    longitude: number;
  };
  imageUrl?: string;
}

export interface User {
  id: string;
  name: string;
  email: string;
  skillLevel: SkillLevel;
  rating?: number;
}

export interface Game {
  id: string;
  title: string;
  description: string;
  startTime: string;
  endTime: string;
  location: Location;
  host: User;
  players: User[];
  maxPlayers: number;
  skillLevel: SkillLevel;
  price: number;
  imageUrl?: string;
  status: GameStatus;
  createdAt: string;
  updatedAt: string;
}

export interface GameFilters {
  skillLevel?: SkillLevel;
  date?: string;
  location?: string;
  status?: GameStatus;
} 