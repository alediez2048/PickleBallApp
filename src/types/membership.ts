export interface MembershipPlan {
  id: string;
  name: string;
  price: number;
  interval?: "month";
  benefits: string[];
  description: string;
}

export type MembershipTier = "drop-in" | "monthly";
