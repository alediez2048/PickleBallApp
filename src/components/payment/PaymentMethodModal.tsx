import React, { useState, useEffect } from "react";
import {
  View,
  StyleSheet,
  Modal,
  TouchableOpacity,
  SafeAreaView,
  Platform,
  ScrollView,
  Alert,
} from "react-native";
import { IconSymbol } from "@/components/ui/IconSymbol";
import { Button } from "@/components/common/ui/Button";
import { MembershipPlan } from "@/types/membership";
import { CardField, useConfirmPayment } from "@stripe/stripe-react-native";
import { ThemedView } from "@/components/common/ThemedView";
import { ThemedText } from "@/components/common/ThemedText";

interface PaymentMethodModalProps {
  visible: boolean;
  onClose: () => void;
  onComplete: () => void;
  selectedPlan: MembershipPlan;
}

export function PaymentMethodModal({
  visible,
  onClose,
  onComplete,
  selectedPlan,
}: PaymentMethodModalProps) {
  const [isLoading, setIsLoading] = useState(false);
  const [isSuccess, setIsSuccess] = useState(false);
  const [cardDetails, setCardDetails] = useState<any>(null);
  const { confirmPayment } = useConfirmPayment();

  useEffect(() => {
    if (visible) {
      setIsSuccess(false);
      setCardDetails(null);
    }
  }, [visible]);

  // TODO: Replace with your backend call to create a PaymentIntent and return clientSecret
  const fetchPaymentIntentClientSecret = async () => {
    // Example: fetch from your Supabase Edge Function or API
    // const res = await fetch('https://your-backend/create-payment-intent', { method: 'POST', body: JSON.stringify({ amount: selectedPlan.price }) });
    // const { clientSecret } = await res.json();
    // return clientSecret;
    return "REPLACE_WITH_CLIENT_SECRET_FROM_BACKEND";
  };

  const handleSubmit = async () => {
    if (Platform.OS !== "web" && (!cardDetails || !cardDetails.complete)) {
      Alert.alert("Invalid Card", "Please enter complete card details.");
      return;
    }
    setIsLoading(true);
    try {
      const clientSecret = await fetchPaymentIntentClientSecret();
      if (
        !clientSecret ||
        clientSecret === "REPLACE_WITH_CLIENT_SECRET_FROM_BACKEND"
      ) {
        setIsLoading(false);
        Alert.alert(
          "Error",
          "You must implement the backend to create a PaymentIntent and return clientSecret."
        );
        return;
      }
      const { error, paymentIntent } = await confirmPayment(clientSecret);
      setIsLoading(false);
      if (error) {
        Alert.alert("Payment failed", error.message);
      } else if (paymentIntent) {
        setIsSuccess(true);
        setTimeout(() => {
          onComplete();
        }, 1500);
      }
    } catch (e: any) {
      setIsLoading(false);
      Alert.alert("Error", e.message || "Something went wrong");
    }
  };

  return (
    <Modal
      visible={visible}
      animationType='slide'
      transparent={true}
      onRequestClose={onClose}
    >
      <SafeAreaView style={styles.container}>
        <ThemedView style={styles.content}>
          <ThemedView style={styles.header}>
            <TouchableOpacity onPress={onClose} style={styles.closeButton}>
              <IconSymbol name='xmark' size={24} color='#666666' />
            </TouchableOpacity>
            <ThemedText style={styles.title}>Payment Method</ThemedText>
            <ThemedText style={styles.subtitle}>
              {isSuccess
                ? "Payment method added successfully!"
                : `Add a payment method for ${selectedPlan.name}`}
            </ThemedText>
          </ThemedView>

          {isSuccess ? (
            <ThemedView style={styles.successContainer}>
              <ThemedView style={styles.successIconContainer}>
                <IconSymbol name='checkmark' size={40} color='#FFFFFF' />
              </ThemedView>
              <ThemedText style={styles.successText}>
                Payment Method Added
              </ThemedText>
              <ThemedText style={styles.successSubtext}>
                Your subscription to {selectedPlan.name} has been activated.
              </ThemedText>
            </ThemedView>
          ) : (
            <ScrollView style={styles.formContainer}>
              <ThemedView style={styles.planSummary}>
                <ThemedText style={styles.planSummaryTitle}>
                  Plan Summary
                </ThemedText>
                <ThemedView style={styles.planDetails}>
                  <ThemedText style={styles.planName}>
                    {selectedPlan.name}
                  </ThemedText>
                  <ThemedText style={styles.planPrice}>
                    ${selectedPlan.price}
                    {selectedPlan.interval ? `/${selectedPlan.interval}` : ""}
                  </ThemedText>
                </ThemedView>
              </ThemedView>

              {Platform.OS === "web" ? (
                <ThemedView style={{ marginVertical: 24 }}>
                  <ThemedText style={{ color: "red", textAlign: "center" }}>
                    Stripe React Native does not support web. Please implement
                    web payment using Stripe.js.
                  </ThemedText>
                </ThemedView>
              ) : (
                <ThemedView style={styles.formGroup}>
                  <ThemedText style={styles.label}>Card Details</ThemedText>
                  <CardField
                    postalCodeEnabled={false}
                    placeholders={{ number: "4242 4242 4242 4242" }}
                    cardStyle={{
                      backgroundColor: "#FFFFFF",
                      textColor: "#000000",
                      borderRadius: 8,
                      borderWidth: 1,
                      borderColor: "#E5E7EB",
                    }}
                    style={{ width: "100%", height: 50, marginVertical: 12 }}
                    onCardChange={setCardDetails}
                  />
                </ThemedView>
              )}

              <Button
                onPress={handleSubmit}
                variant='primary'
                style={styles.submitButton}
                disabled={isLoading}
              >
                {isLoading ? "Processing..." : "Add Payment Method"}
              </Button>
            </ScrollView>
          )}
        </ThemedView>
      </SafeAreaView>
    </Modal>
  );
}

const styles = StyleSheet.create({
  container: {
    flex: 1,
    backgroundColor: "rgba(0, 0, 0, 0.5)",
  },
  content: {
    flex: 1,
    backgroundColor: "#FFFFFF",
    borderTopLeftRadius: 20,
    borderTopRightRadius: 20,
    marginTop: 60,
  },
  header: {
    padding: 20,
    borderBottomWidth: 1,
    borderBottomColor: "#E5E7EB",
  },
  closeButton: {
    position: "absolute",
    top: 20,
    right: 20,
    zIndex: 1,
  },
  title: {
    fontSize: 24,
    fontWeight: "bold",
    color: "#000000",
    marginBottom: 8,
  },
  subtitle: {
    fontSize: 16,
    color: "#666666",
  },
  formContainer: {
    padding: 20,
  },
  planSummary: {
    backgroundColor: "#f8f9fa",
    borderRadius: 8,
    padding: 16,
    marginBottom: 24,
  },
  planSummaryTitle: {
    fontSize: 16,
    fontWeight: "600",
    marginBottom: 8,
    color: "#333333",
  },
  planDetails: {
    flexDirection: "row",
    justifyContent: "space-between",
    alignItems: "center",
  },
  planName: {
    fontSize: 18,
    fontWeight: "600",
    color: "#000000",
    marginBottom: 4,
  },
  planPrice: {
    fontSize: 24,
    fontWeight: "bold",
    color: "#000000",
  },
  formGroup: {
    marginBottom: 16,
  },
  label: {
    fontSize: 14,
    fontWeight: "500",
    color: "#333333",
    marginBottom: 8,
  },
  submitButton: {
    marginTop: 24,
  },
  successContainer: {
    alignItems: "center",
    justifyContent: "center",
    padding: 24,
  },
  successIconContainer: {
    width: 80,
    height: 80,
    borderRadius: 40,
    backgroundColor: "#4CAF50",
    alignItems: "center",
    justifyContent: "center",
    marginBottom: 24,
  },
  successText: {
    fontSize: 24,
    fontWeight: "bold",
    color: "#333333",
    marginBottom: 8,
  },
  successSubtext: {
    fontSize: 16,
    color: "#666666",
    textAlign: "center",
  },
});
