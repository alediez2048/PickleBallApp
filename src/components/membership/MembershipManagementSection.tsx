import React, { useState } from "react";
import { StyleSheet, Alert } from "react-native";
import { IconSymbol } from "@/components/common/IconSymbol";
import { Button } from "@/components/common/Button";
import { MembershipPlanModal } from "./MembershipPlanModal";
import { PaymentMethodModal } from "@/components/payment/PaymentMethodModal";
import { MembershipPlan } from "@/types/membership";
import { ThemedText } from "@/components/common/ThemedText";
import { ThemedView } from "@/components/common/ThemedView";
import { useAuth } from "@/contexts/AuthContext";

const MEMBERSHIP_PLANS: MembershipPlan[] = [
  {
    id: "drop-in",
    name: "Drop-In Pass",
    price: 10,
    description: "Perfect for occasional players",
    benefits: [
      "Single game access",
      "No commitment required",
      "Full access to game features",
      "Cancel anytime",
    ],
  },
  {
    id: "monthly",
    name: "Monthly Membership",
    price: 50,
    interval: "month",
    description: "Best value for regular players",
    benefits: [
      "Unlimited game access",
      "Priority booking",
      "Member-only events",
      "Exclusive discounts",
      "Cancel anytime",
    ],
  },
];

interface MembershipManagementSectionProps {
  currentPlan?: MembershipPlan;
  onUpdatePlan: (plan: MembershipPlan) => void;
}

export function MembershipManagementSection({
  currentPlan,
  onUpdatePlan,
}: MembershipManagementSectionProps) {
  const { user } = useAuth();
  const [showPlanModal, setShowPlanModal] = useState(false);
  const [showPaymentModal, setShowPaymentModal] = useState(false);
  const [selectedPlan, setSelectedPlan] = useState<MembershipPlan | null>(null);

  // Mock payment method for testing
  const mockPaymentMethod = {
    id: "mock-payment-1",
    last4: "4242",
    brand: "Visa",
    expiryMonth: "12",
    expiryYear: "25",
    isDefault: true,
  };

  // Check if user has a payment method
  const hasPaymentMethod = true; // For testing

  // Get default payment method if available
  const defaultPaymentMethod = mockPaymentMethod; // For testing

  const handlePlanSelect = (plan: MembershipPlan) => {
    setSelectedPlan(plan);

    // If selecting the same plan, just close the modal
    if (currentPlan && plan.id === currentPlan.id) {
      Alert.alert(
        "Current Plan",
        `You are already subscribed to the ${plan.name} plan.`,
        [
          {
            text: "OK",
            onPress: () => setShowPlanModal(false),
          },
        ]
      );
      return;
    }

    // If changing between paid plans
    if (currentPlan) {
      Alert.alert(
        "Change Plan",
        `Are you sure you want to change to the ${
          plan.name
        } plan? Your billing will be updated to ${formatPrice(
          plan.price,
          plan.interval
        )}.`,
        [
          {
            text: "Cancel",
            style: "cancel",
          },
          {
            text: "Change Plan",
            onPress: () => {
              setShowPlanModal(false);
              onUpdatePlan(plan);
            },
          },
        ]
      );
      return;
    }

    // If no current plan, show payment modal for new plan
    if (hasPaymentMethod) {
      Alert.alert(
        "Confirm Plan Selection",
        `Are you sure you want to select the ${
          plan.name
        } plan? You will be charged ${formatPrice(plan.price, plan.interval)}.`,
        [
          {
            text: "Cancel",
            style: "cancel",
          },
          {
            text: "Confirm",
            onPress: () => {
              setShowPlanModal(false);
              onUpdatePlan(plan);
            },
          },
        ]
      );
    } else {
      // If no payment method, show payment modal
      setShowPlanModal(false);
      setShowPaymentModal(true);
    }
  };

  const handlePaymentComplete = () => {
    setShowPaymentModal(false);
    if (selectedPlan) {
      onUpdatePlan(selectedPlan);
    }
  };

  const formatPrice = (price: number, interval?: string) => {
    return `$${price}${interval ? `/${interval}` : ""}`;
  };

  // Determine the primary action based on current plan
  const getPrimaryAction = () => {
    if (!currentPlan) {
      return (
        <Button
          variant="primary"
          onPress={() => setShowPlanModal(true)}
          style={styles.actionButton}
        >
          Select a Plan
        </Button>
      );
    }

    return (
      <ThemedView style={styles.actionButtonsRow} type="none">
        <Button
          variant="outline"
          onPress={() => setShowPlanModal(true)}
          style={[styles.actionButton, styles.actionButtonHalf]}
        >
          Change Plan
        </Button>
        <Button
          variant="outline"
          onPress={() => {
            Alert.alert(
              "Update Payment Method",
              "Do you want to update your payment method?",
              [
                {
                  text: "Cancel",
                  style: "cancel",
                },
                {
                  text: "Update",
                  onPress: () => {
                    setSelectedPlan(currentPlan);
                    setShowPaymentModal(true);
                  },
                },
              ]
            );
          }}
          style={[styles.actionButton, styles.actionButtonHalf]}
        >
          Update Payment
        </Button>
      </ThemedView>
    );
  };

  return (
    <ThemedView style={styles.container}>
      {currentPlan ? (
        <ThemedView style={styles.planCard} type="card">
          <ThemedView style={styles.planHeader} type="none">
            <ThemedView style={styles.planTitleContainer} type="none">
              <ThemedText style={styles.planName} type="subtitle">
                {currentPlan.name}
              </ThemedText>
              <ThemedText style={styles.planPrice} type="bold">
                {formatPrice(currentPlan.price, currentPlan.interval)}
              </ThemedText>
            </ThemedView>
          </ThemedView>
          <ThemedText style={styles.planDescription} type="paragraph">
            {currentPlan.description}
          </ThemedText>
          <ThemedView style={styles.divider} type="none" />
          <ThemedView style={styles.benefitsContainer} type="none">
            {currentPlan.benefits.map((benefit, idx) => (
              <ThemedView key={idx} style={styles.benefitRow} type="none">
                <ThemedText style={styles.benefitText} type="default">
                  {benefit}
                </ThemedText>
              </ThemedView>
            ))}
          </ThemedView>
          <ThemedView style={styles.divider} type="none" />
          {defaultPaymentMethod && (
            <ThemedView style={styles.paymentMethodContainer} type="bordered">
              <ThemedView style={styles.paymentMethodRow} type="none">
                <ThemedText style={styles.paymentText} type="label">
                  {defaultPaymentMethod.brand} ****{defaultPaymentMethod.last4}
                </ThemedText>
                <ThemedText style={styles.paymentExpiry} type="caption">
                  Exp: {defaultPaymentMethod.expiryMonth}/
                  {defaultPaymentMethod.expiryYear}
                </ThemedText>
              </ThemedView>
            </ThemedView>
          )}
          <ThemedView style={styles.actionsContainer} type="none">
            {getPrimaryAction()}
          </ThemedView>
        </ThemedView>
      ) : (
        <ThemedView
          style={styles.emptyState}
          borderColorType="primary"
          borderWidth={2}
        >
          <ThemedText type="value">No Membership Plan Selected</ThemedText>
          <ThemedText style={styles.emptySubtext} type="emptyStateText">
            Choose a plan to get started and unlock all features.
          </ThemedText>
          <Button
            variant="primary"
            size="small"
            onPress={() => setShowPlanModal(true)}
            style={styles.emptyStateButton}
          >
            Select a Plan
          </Button>
        </ThemedView>
      )}
      <MembershipPlanModal
        visible={showPlanModal}
        onClose={() => setShowPlanModal(false)}
        onSelectPlan={handlePlanSelect}
        currentPlanId={currentPlan?.id}
      />
      {selectedPlan && (
        <PaymentMethodModal
          visible={showPaymentModal}
          onClose={() => setShowPaymentModal(false)}
          onComplete={handlePaymentComplete}
          selectedPlan={selectedPlan}
        />
      )}
    </ThemedView>
  );
}

const styles = StyleSheet.create({
  container: {
    width: "100%",
  },
  planCard: {
    borderRadius: 12,
    padding: 16,
    shadowColor: "#000",
    shadowOffset: { width: 0, height: 2 },
    shadowOpacity: 0.1,
    shadowRadius: 4,
    elevation: 2,
    overflow: "hidden",
  },
  planHeader: {
    flexDirection: "row",
    justifyContent: "space-between",
    alignItems: "center",
    marginBottom: 12,
    flexWrap: "wrap",
  },
  planTitleContainer: {
    flexDirection: "row",
    alignItems: "center",
    flexShrink: 1,
    marginRight: 8,
  },
  planName: {
    fontWeight: "600",
    color: "#666666",
    fontSize: 18,
    flexShrink: 1,
  },
  planPrice: {
    fontSize: 20,
    fontWeight: "bold",
    color: "#4CAF50",
  },
  planDescription: {
    color: "#666666",
    marginBottom: 12,
    lineHeight: 20,
  },
  divider: {
    height: 1,
    backgroundColor: "#e0e0e0",
    marginVertical: 12,
  },
  benefitsContainer: {
    gap: 10,
    marginBottom: 12,
  },
  benefitRow: {
    flexDirection: "row",
    alignItems: "center",
    gap: 10,
  },
  benefitText: {
    fontSize: 14,
    color: "#333333",
    flex: 1,
  },
  paymentMethodContainer: {
    marginBottom: 16,
    borderRadius: 8,
    padding: 12,
  },
  paymentMethodRow: {
    flexDirection: "row",
    alignItems: "center",
    marginBottom: 4,
  },
  paymentIcon: {
    marginRight: 8,
  },
  paymentText: {
    fontSize: 14,
    color: "#333333",
    fontWeight: "500",
  },
  paymentExpiry: {
    color: "#666666",
    marginLeft: 24,
  },
  actionsContainer: {
    marginTop: 16,
  },
  actionButton: {
    width: "100%",
  },
  actionButtonsRow: {
    flexDirection: "row",
    justifyContent: "space-between",
    gap: 12,
  },
  actionButtonHalf: {
    flex: 1,
  },
  emptyState: {
    alignItems: "center",
    padding: 24,
    borderRadius: 12,
    shadowColor: "#000",
    shadowOffset: { width: 0, height: 2 },
    shadowOpacity: 0.1,
    shadowRadius: 4,
    elevation: 2,
  },
  emptyIconContainer: {
    backgroundColor: "#4CAF50",
    width: 56,
    height: 56,
    borderRadius: 28,
    justifyContent: "center",
    alignItems: "center",
    marginBottom: 16,
  },
  emptyText: {
    fontSize: 18,
    fontWeight: "600",
    marginBottom: 8,
    color: "#333333",
  },
  emptySubtext: {
    textAlign: "center",
    marginTop: 16,
    marginBottom: 16,
    lineHeight: 20,
  },
  emptyStateButton: {
    marginTop: 8,
    minWidth: 200,
  },
});
