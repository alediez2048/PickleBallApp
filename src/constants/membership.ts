import { MembershipPlan } from "@/types/membership";

export const MEMBERSHIP_PLANS: MembershipPlan[] = [
  {
    id: "drop-in",
    name: "Drop-In Pass",
    price: 10,
    description: "Perfect for occasional players",
    benefits: [
      "Single game access",
      "No commitment required",
      "Full access to game features",
      "Cancel anytime",
    ],
  },
  {
    id: "monthly",
    name: "Monthly Membership",
    price: 50,
    interval: "month",
    description: "Best value for regular players",
    benefits: [
      "Unlimited game access",
      "Priority booking",
      "Member-only events",
      "Exclusive discounts",
      "Cancel anytime",
    ],
  },
];