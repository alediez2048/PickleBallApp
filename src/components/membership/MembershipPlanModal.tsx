import React from "react";
import {
  StyleSheet,
  Modal,
  TouchableOpacity,
  SafeAreaView,
  Platform,
  ScrollView,
} from "react-native";
import { IconSymbol } from "@/components/common/IconSymbol";
import { Button } from "@/components/common/Button";
import { ThemedView } from "@/components/common/ThemedView";
import { ThemedText } from "@/components/common/ThemedText";
import { useTheme } from "@/contexts/ThemeContext";
import { MembershipPlan } from "@/types/membership";
import { MEMBERSHIP_PLANS } from "@/constants/membership";

interface MembershipPlanModalProps {
  visible: boolean;
  onClose: () => void;
  onSelectPlan: (plan: MembershipPlan) => void;
  currentPlanId?: string;
}

export function MembershipPlanModal({
  visible,
  onClose,
  onSelectPlan,
  currentPlanId,
}: MembershipPlanModalProps) {
  const { colors } = useTheme();
  return (
    <Modal
      visible={visible}
      animationType="slide"
      transparent={true}
      onRequestClose={onClose}
    >
      <SafeAreaView style={styles.container}>
        <ThemedView style={styles.content}>
          <ThemedView style={styles.header}>
            <TouchableOpacity onPress={onClose} style={styles.closeButton}>
              <IconSymbol name="xmark" size={24} />
            </TouchableOpacity>
            <ThemedText type="title">Choose Your Plan</ThemedText>
            <ThemedText type="miniSubtitle" colorType="label">
              Select a membership plan that fits your needs
            </ThemedText>
          </ThemedView>

          <ScrollView style={styles.plansContainer}>
            {MEMBERSHIP_PLANS.map((plan) => (
              <TouchableOpacity
                key={plan.id}
                style={[
                  styles.planCard,
                  { borderColor: colors.text },
                  currentPlanId === plan.id && styles.currentPlanCard,
                ]}
                onPress={() => onSelectPlan(plan)}
              >
                {currentPlanId === plan.id && (
                  <ThemedView
                    style={styles.currentPlanBadge}
                    colorType="primary"
                  >
                    <ThemedText
                      type="badge"
                      colorType="white"
                      style={styles.currentPlanBadgeText}
                    >
                      Current Plan
                    </ThemedText>
                  </ThemedView>
                )}

                <ThemedView style={styles.planHeader}>
                  <ThemedText>{plan.name}</ThemedText>
                  <ThemedView style={styles.priceContainer}>
                    <ThemedText type="title" colorType="primary">
                      ${plan.price}
                    </ThemedText>
                    {plan.interval && (
                      <ThemedText colorType="label">
                        / {plan.interval}
                      </ThemedText>
                    )}
                  </ThemedView>
                </ThemedView>

                <ThemedText>{plan.description}</ThemedText>

                <ThemedView style={styles.benefitsContainer}>
                  {plan.benefits.map((benefit, index) => (
                    <ThemedView key={index} style={styles.benefitRow}>
                      <IconSymbol
                        name="checkmark"
                        size={18}
                        color={colors.success}
                        style={styles.benefitIcon}
                      />
                      <ThemedText type="label">{benefit}</ThemedText>
                    </ThemedView>
                  ))}
                </ThemedView>

                <Button
                  variant={currentPlanId === plan.id ? "outline" : "primary"}
                  style={styles.selectButton}
                  onPress={() => onSelectPlan(plan)}
                >
                  {currentPlanId === plan.id ? "Current Plan" : "Select Plan"}
                </Button>
              </TouchableOpacity>
            ))}
          </ScrollView>
        </ThemedView>
      </SafeAreaView>
    </Modal>
  );
}

const styles = StyleSheet.create({
  container: {
    flex: 1,
  },
  content: {
    flex: 1,
    borderTopLeftRadius: 20,
    borderTopRightRadius: 20,
    marginTop: Platform.OS === "ios" ? 50 : 30,
    overflow: "hidden",
  },
  header: {
    padding: 20,
    borderBottomWidth: 1,
  },
  closeButton: {
    position: "absolute",
    top: 20,
    right: 20,
    zIndex: 1,
  },
  plansContainer: {
    padding: 20,
  },
  planCard: {
    borderRadius: 12,
    padding: 10,
    marginBottom: 20,
    borderWidth: 2,
  },
  currentPlanCard: {
    borderColor: "#4CAF50",
    borderWidth: 2,
  },
  currentPlanBadge: {
    backgroundColor: "#00f308",
    paddingHorizontal: 12,
    paddingVertical: 6,
    borderRadius: 20,
    position: "absolute",
    top: -12,
    right: 20,
    zIndex: 1,
  },
  currentPlanBadgeText: {
    color: "white",
    fontSize: 12,
    fontWeight: "bold",
  },
  planHeader: {
    marginBottom: 12,
  },
  priceContainer: {
    flexDirection: "row",
    alignItems: "baseline",
  },
  benefitsContainer: {
    marginBottom: 20,
  },
  benefitRow: {
    flexDirection: "row",
    alignItems: "center",
    marginVertical: 4,
  },
  benefitIcon: {
    marginRight: 10,
  },
  selectButton: {
    width: "100%",
  },
});
