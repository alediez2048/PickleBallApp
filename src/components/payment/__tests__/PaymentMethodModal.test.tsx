import React from 'react';
import renderer from 'react-test-renderer';
import { PaymentMethodModal } from '../PaymentMethodModal';
import { MembershipPlan } from '@/types/membership';

// Mock the PaymentMethodForm component to simplify testing
jest.mock('../PaymentMethodForm', () => ({
  PaymentMethodForm: () => "MockPaymentMethodForm"
}));

describe('PaymentMethodModal', () => {
  const mockOnClose = jest.fn();
  const mockOnComplete = jest.fn();
  const mockPlan: MembershipPlan = {
    id: 'test-plan',
    name: 'Test Plan',
    price: 10,
    interval: 'month',
    description: 'Test description',
    benefits: ['Benefit 1', 'Benefit 2'],
  };

  beforeEach(() => {
    jest.clearAllMocks();
  });

  it('renders correctly with selected plan', () => {
    const tree = renderer
      .create(
        <PaymentMethodModal
          visible={true}
          onClose={mockOnClose}
          onComplete={mockOnComplete}
          selectedPlan={mockPlan}
        />
      )
      .toJSON();
    expect(tree).toMatchSnapshot();
  });

  it('renders correctly when not visible', () => {
    const tree = renderer
      .create(
        <PaymentMethodModal
          visible={false}
          onClose={mockOnClose}
          onComplete={mockOnComplete}
          selectedPlan={mockPlan}
        />
      )
      .toJSON();
    expect(tree).toMatchSnapshot();
  });
}); 