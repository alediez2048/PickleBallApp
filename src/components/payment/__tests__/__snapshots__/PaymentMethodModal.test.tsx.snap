// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`PaymentMethodModal renders correctly when not visible 1`] = `null`;

exports[`PaymentMethodModal renders correctly with selected plan 1`] = `null`;
